src/plugins/ui_actions/public/context_menu/build_eui_context_menu_panels.tsx

```diff
@@ -211,10 +228,17 @@ export async function buildContextMenuForActions({
   wrapMainPanelItemsIntoSubmenu(panels, 'mainMenu');
 
   for (const panel of Object.values(panels)) {
+    // If the panel is a root-level panel, such as a group-based panel,
+    // then create mainMenu item for this panel
     if (panel._level === 0) {
-      // TODO: Add separator line here once it is available in EUI.
-      // See https://github.com/elastic/eui/pull/4018
-      if (panel.items.length > 3) {
+      // Add separator with unique key if needed
+      if (panels.mainMenu.items.length) {
+        panels.mainMenu.items.push({ isSeparator: true, key: `${panel.id}separator` });
+      }
+
+      // If a panel has more than one child, then allow item's to be grouped
+      // and link to it in the mainMenu. Otherwise, flatten the group.
+      if (panel.items.length > 1) {
```

Comment:
Can you add a unit test for this?

Reply:
Yes, will add this soon and comment here when completed 👍

Reply:
Test added ✅